python/dgl/graph.py

```diff
@@ -1244,6 +1279,112 @@ def remove_edges(self, eids):
 
         self._graph = sgi.graph
 
+    @property
+    def parent_nid(self):
+        """Get the parent node ids.
+
+        The returned tensor can be used as a map from the node id
+        in this subgraph to the node id in the parent graph.
+
+        Returns
+        -------
+        Tensor
+            The parent node id array.
+        """
+        if self._parent is None:
+            raise DGLError("We only support parent_nid for subgraphs.")
+        return self.ndata[NID]
+
+    @property
+    def parent_eid(self):
+        """Get the parent edge ids.
+
+        The returned tensor can be used as a map from the edge id
+        in this subgraph to the edge id in the parent graph.
+
+        Returns
+        -------
+        Tensor
+            The parent edge id array.
+        """
+        if self._parent is None:
+            raise DGLError("We only support parent_eid for subgraphs.")
+        return self.edata[EID]
+
+    def copy_to_parent(self, inplace=False):
+        """Write node/edge features to the parent graph.
+
+        Parameters
+        ----------
+        inplace : bool
+            If true, use inplace write (no gradient but faster)
+        """
+        if self._parent is None:
+            raise DGLError("We only support copy_to_parent for subgraphs.")
+        nids = self.ndata.pop(NID)
```

Comment:
@jermainewang do you have any suggestions on improving this? We don't want the EID/NID field of subgraph to pollute parent graph's frame.

Reply:
I think your solution is fine. I don't have a better solution at the moment.